### ISSUE TITLE
jetty-12 ee9 ServerPush failures

### ISSUE THREAD

**Issue by janbartel** [eclipse/jetty.project#9766]
### Tck test: com/sun/ts/tests/servlet/spec/serverpush/Client.java.serverPushSessionTest2

This test sends an http/2 request to the server to create a session and push the index.html page back to the client. The test fails during the handling of `pb.push()` by never returning to the client.

There is one related `TODO` in the code:  https://github.com/eclipse/jetty.project/blob/jetty-12.0.x/jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/PushBuilderImpl.java#L177

The client code looks like:
``` java
        requestURI = "http://" + hostname + ":" + portnum + getContextRoot() + "/TestServlet5";
        Map<String, String> headers = new HashMap<>();
        CookieManager cm = new CookieManager();
        cm.setCookiePolicy(CookiePolicy.ACCEPT_ALL);
        List<HttpResponse<String>> responses = sendRequest(headers, null, cm);

        List<HttpCookie> cookies = cm.getCookieStore().get(new URI(
                "http://" + hostname + ":" + portnum + getContextRoot() + "/index.html"));
        for (HttpCookie cookie : cookies) {
            if ("JSESSIONID".equals(cookie.getName())) {
                pass = true;
                break;
            }
        }
```

The server code looks like:
``` java
    HttpSession session = req.getSession(true);
    pw.print(session.getId());
    PushBuilder pb = req.newPushBuilder();
    pb.path("index.html");
    pb.push();
```

### Tck test: com/sun/ts/tests/servlet/spec/serverpush/Client.java.serverPushCookieTest

This test sends an http/2 request to the server, which creates a session plus some extra cookies and pushes the response back to the client. The test never finishes somewhere inside the handling of the `pb.push()` call.

The client code looks like:
``` java
        requestURI = "http://" + hostname + ":" + portnum + getContextRoot() +
                "/TestServlet4";
        Map<String, String> headers = new HashMap<>();
        headers.put("foo", "bar");
        CookieManager cm = new CookieManager();
        List<HttpResponse<String>> responses = sendRequest(headers, null, cm);
        verifyResponses(responses, new String[] { "add cookies [foo,bar] [baz,qux] to response", "INDEX from index.html" });
```

The server code looks like:
``` java
    Cookie cookie1 = new Cookie("foo", "bar");
    cookie1.setMaxAge(1000);
    resp.addCookie(cookie1);

    Cookie cookie2 = new Cookie("baz", "qux");
    cookie2.setMaxAge(-1);
    resp.addCookie(cookie2);
    pw.println("add cookies [foo,bar] [baz,qux] to response");

    HttpSession session = req.getSession(true);
    pw.println("create session: " + session);

    PushBuilder pb = req.newPushBuilder();
    pw.println("Cookie header in PushBuilder: " + pb.getHeader("Cookie"));
    pb.path("index.html");
    pb.push();
```

**Comment by sbordet** [comment]
@janbartel for the first one, I'd say that should be enough to restore the TODO with:

```java
            if (_request.isRequestedSessionIdFromURL())
                param = "jsessionid=" + _sessionId;
             else
                 _fields.put(HttpHeader.COOKIE,"JSESSIONID=" + _sessionId);
```

**Comment by sbordet** [comment]
@janbartel the second looks like a broken test to me.
Given that the client does not send any cookie, the server should just push `index.html` with no cookies, so I'm not sure what's going on (as other simple pushes do not block like you report).

**Comment by janbartel** [comment]
Fixed via #9776
